packages/core/base/src/standard.ts

```diff
@@ -0,0 +1,33 @@
+import type {
+    SolanaSignAndSendTransactionFeature,
+    SolanaSignMessageFeature,
+    SolanaSignTransactionFeature,
+} from '@solana/wallet-standard-features';
+import type { Wallet as StandardWallet, WalletWithFeatures as StandardWalletWithFeatures } from '@wallet-standard/base';
+import type { ConnectFeature, DisconnectFeature, EventsFeature } from '@wallet-standard/features';
+import type { WalletAdapter, WalletAdapterProps } from './adapter.js';
+
+export type WalletAdapterCompatibleStandardWallet = StandardWalletWithFeatures<
+    ConnectFeature &
+        EventsFeature &
+        (SolanaSignAndSendTransactionFeature | SolanaSignTransactionFeature) &
+        (DisconnectFeature | SolanaSignMessageFeature | object)
+>;
+
+export interface StandardWalletAdapterProps<Name extends string = string> extends WalletAdapterProps<Name> {
+    wallet: WalletAdapterCompatibleStandardWallet;
+    standard: true;
+}
+
+export type StandardWalletAdapter<Name extends string = string> = WalletAdapter<Name> &
+    StandardWalletAdapterProps<Name>;
+
+export function isWalletAdapterCompatibleStandardWallet(
+    wallet: StandardWallet
+): wallet is WalletAdapterCompatibleStandardWallet {
+    return (
+        'standard:connect' in wallet.features &&
+        'standard:events' in wallet.features &&
+        ('solana:signAndSendTransaction' in wallet.features || 'solana:signTransaction' in wallet.features)
```

Comment:
Is there a tension between naming this ‘is standard wallet’ and having Solana specific checks in here? Should this method be more like ‘is this a Solana standard wallet?’

Reply:
I'm not sure I understand the question. We know `wallet` is a Standard Wallet, but we don't know if it's Wallet Adapter-compatible. A hypothetical "Solana" Standard Wallet (let's say, one that has `solana:` chains and features) would be incompatible with Wallet Adapter if it lacks these features.

Reply:
Haha, all good. I really didn't provide any context when asking for review on this...